### PR TITLE
wasi-nn: support multiple TFLite models

### PR DESCRIPTION
**Pull request by tonibofarull**
Remove restrictions:

```
* Only 1 WASM app at a time.
* Only 1 model at a time.
    * `graph` and `graph-execution-context` are ignored.
```

Check [here](https://github.com/bytecodealliance/wasm-micro-runtime/blob/e8d718096dc56d4b1aa66ec6cd04d6024ca1c6e2/core/iwasm/libraries/wasi-nn/README.md).

### CODE REVIEW

File: core/iwasm/libraries/wasi-nn/src/wasi_nn_private.h [hunk 1]
```diff
@@ -11,6 +11,7 @@
 typedef struct {
     bool is_initialized;
     graph_encoding current_encoding;
+    void *tflite_ctx;
```
Comment (tonibofarull) [inline core/iwasm/libraries/wasi-nn/src/wasi_nn_private.h:14]: `void *` to hide C++ specification

File: core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp [hunk 2]
```diff
@@ -16,25 +16,68 @@
...
+static bool
+initialize_g(TFLiteContext *tfl_ctx, graph *g)
+{
+    os_mutex_lock(&tfl_ctx->g_lock);
+    if (tfl_ctx->current_models == MAX_INST_MODELS) {
+        os_mutex_unlock(&tfl_ctx->g_lock);
+        return false;
+    }
+    *g = tfl_ctx->current_models++;
+    os_mutex_unlock(&tfl_ctx->g_lock);
```
Comment (tonibofarull) [inline core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp:57]: Is there an atomic instruction in WAMR?

**Review by wenyongh — COMMENTED** [review]
LGTM

**Comment by wenyongh** [comment]
@tonibofarull We fixed some issues which make CI report errors, could you rebase the PR to run CIs again? Thanks.

**Comment by tonibofarull** [comment]
> @tonibofarull We fixed some issues which made CI report errors, could you rebase the PR to run CIs again? Thanks.

Done!

**Pull request merged**